src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1240 @@
+
+  def generate_FDB_6_default_config(self, filename):
+    """Generates the WinnForum configuration for FDB.6"""
+
+    # Load devices info
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # Load grant requests
+    grant_g1_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    # Set the grant frequency to overlap with the FSS 'KA413' which is 3625-4200 MHz.
+    grant_g1_a['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3650000000,
+        'highFrequency': 3660000000
+    }
+    grant_g2_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    # Set the grant frequency to overlap with the FSS 'KA413' which is 3625-4200 MHz.
+    grant_g2_a['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3675000000,
+        'highFrequency': 3685000000
+    }
+
+    grant_g1_b = json.load(
+                open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    # Set the grant frequency to overlap with the FSS 'E000306' which is 3625-3700 MHz.
+    grant_g1_b['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3660000000,
+        'highFrequency': 3670000000
+    }
+    grant_g2_b = json.load(
+                open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    # Set the grant frequency to overlap with the FSS 'E000306' which is 3625-3700 MHz.
+    grant_g2_b['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3685000000,
+        'highFrequency': 3695000000
+    }
+
+    # Update the location 'X' of CBSD devices to be near FSS sites
+    # 'KA413' at Albright,WV
+    device_a['installationParam']['latitude'] = 39.57327
+    device_a['installationParam']['longitude'] = -79.61903
+
+    # 'E000306' at Andover,ME
+    device_b['installationParam']['latitude'] = 44.63367
+    device_b['installationParam']['longitude'] = -70.69758
+
+    # Creating conditionals for Cat B devices
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+    del device_b['measCapability']
+
+    conditionals = [conditionals_b]
+
+    # Create the actual config.
+    config = {
+        'registrationRequest': [device_a, device_b],
+        'grantRequests': [
+            [grant_g1_a, grant_g1_b],
+            [grant_g2_a, grant_g2_b]],
+        'expectedResponseCodes': [(400,), (0,)],
```

Comment:
If I read this consistent with other test case implementations, I would take this to mean "device_a must get responseCode 400 during both CHECKs and device_b must get responseCode 0 during both CHECKs". But in fact we need a 2D matrix: device_a could be expected to get 400 during the first check and 0 during the second check, and device_b could have the opposite (or same!) expectation.

Reply:
FDB.6     Added a 2-D matrix for response codes in config data structure:     'expectedResponseCodes': [ [(400,), (0, 400)],  [(0,), (0, 400)]],     Both the lists correspond to the response codes expected in grants 'G1' and   'G2' respectively by devices 'A' and 'B'.